### PR TITLE
data: always use arrays for `queryKey` in `use(Query|Mutation)`

### PR DESCRIPTION
**Pull request by flootr**
Related to #74381

This PR is part of the effort to upgrade `react-query` to the next version. `queryKey` now always has to be an array and doesn't accept a string anymore.

## Proposed Changes

* Convert all string `queryKey`s to arrays.

## Testing Instructions

* This should have no effect on the production build but smoke-testing is probably a good idea
* All checks should be green

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/components/language-picker/site-language-picker.jsx [hunk 1]
```diff
@@ -35,7 +35,7 @@ const SiteLanguagePicker = ( { languages: origLanguages, ...restProps } ) => {
 		error,
 		isLoading,
 	} = useQuery(
-		'wporg-translations-' + wpVersion,
+		[ 'wporg-translations-' + wpVersion ],
```
Comment (tyxla) [inline client/components/language-picker/site-language-picker.jsx]: Noting that now when we always use an array for query keys, we could have the library actually take care of [hashing the key](https://tanstack.com/query/v4/docs/react/guides/query-keys#query-keys-are-hashed-deterministically). Meaning, we could alter this to:

```suggestion
		[ 'wporg-translations', wpVersion ],
```

That applies to all query keys that we're currently manually hashing.

File: client/data/marketplace/utils.ts [hunk 2]
```diff
@@ -18,5 +18,5 @@ export const getPluginsListKey = (
 		options.tag && ! options.searchTerm ? options.tag : '',
 	];
 
-	return [ PLUGINS_CACHE_VERSION.toString(), key, ...keyParams ];
+	return [ PLUGINS_CACHE_VERSION.toString(), ...key, ...keyParams ];
```
Comment (tyxla) [inline client/data/marketplace/utils.ts]: I know this was here before, but I wonder if the `key` part should be at the beginning and the cache version would be afterwards.

**Review by tyxla — APPROVED** [review]
LGTM 👍 

I'd personally suggest for query keys with a dynamic part that the dynamic part would be a separate array item, but that's optional and could be followed up with in another PR.

Thanks 🚀

File: client/data/geo/use-geolocation-query.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import { useQuery } from 'react-query';
 
 export const useGeoLocationQuery = () =>
-	useQuery( 'geo', () =>
+	useQuery( [ 'geo' ], () =>
```
Comment (sirbrillig) [inline client/data/geo/use-geolocation-query.js]: I'll take care of this in https://github.com/Automattic/wp-calypso/pull/75401

File: client/components/language-picker/site-language-picker.jsx (hunk 1 above)
Comment (flootr) [reply client/components/language-picker/site-language-picker.jsx]: Agreed 👍🏻 I'll add a commit which takes care of these.

File: client/data/marketplace/utils.ts (hunk 2 above)
Comment (flootr) [reply client/data/marketplace/utils.ts]: Good point, I'll double-check and add a commit.

Comment (tyxla) [reply client/data/marketplace/utils.ts]: Sounds good. Can be done in another PR if you don't want to clutter this one.

File: client/components/language-picker/site-language-picker.jsx (hunk 1 above)
Comment (tyxla) [reply client/components/language-picker/site-language-picker.jsx]: That totally makes sense to be another PR, FWIW.

Comment (flootr) [reply client/components/language-picker/site-language-picker.jsx]: Added here https://github.com/Automattic/wp-calypso/pull/74720/commits/01d47ac5b2b5d68cca4ae4f05437774fd94779b4

File: client/data/marketplace/utils.ts (hunk 2 above)
Comment (flootr) [reply client/data/marketplace/utils.ts]: See https://github.com/Automattic/wp-calypso/pull/74720/commits/d8ddad71b60b687e64b1dc92f1d7f2ecdabc30c0

**Review by tyxla — APPROVED** [review]
LGTM, thanks 🚀

**Pull request merged**